Adding the first 4 tasks in text to video about file

snknitin opened the pull request:
This is a PR that adds description to the text-to-video task page. it includes  

Task variants 
Use cases 
Most Used Dataset for the Task 
Most Used model for this task

snknitin commented:
@merveenoyan  @osanseviero  This is related to the recent [PR](https://github.com/huggingface/hub-docs/issues/789). Please guide me on the steps for merge and if there is any changes required. I don't see most common datasets or models on other tasks, so i included it in the about.md . I wasn't sure where to add them

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+## Use Cases
+
+Text-to-Video (T2V) models can be used in any application that requires generating a sequence of images from text descriptions, that are both temporally and spatially consistent. Text-to-video AI models are a powerful tool that can be used for a variety of purposes across different domains. As the technology continues to develop, T2V models will become even more powerful and versatile
```
``
```suggestion
Text-to-video models can be used in any application that requires generating temporally and spatially consistent sequence of images from text.  

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Script based video generation
+
+Creating shortform video content from the provided text script. T2V models can be used to create marketing videos that are more engaging and informative. For example, a company could use a text-to-video AI model to create a video that explains how their product works or that shows how their product can solve a customer's problem.
```
```suggestion
Text-to-video models can be used to create short-form video content from a provided text script. These models can be used to create engaging and informative marketing videos. For example, a company could use a text-to-video model to create a video that explains how their product works or that shows how their product can solve a customer's problem.

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Disaster relief and Safety protocols
+
+T2V can be used to generate videos that can be used to help with disaster relief efforts. For example, a text to video model could be used to generate videos that show people how to stay safe during a hurricane or how to find food and water after a natural disaster.
```
```suggestion
Text-to-video models can be used to generate videos that show people how to stay safe during a hurricane or how to find food and water after a natural disaster.

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Medical research
+
+T2V models can also be used to generate videos that can be used to help with medical research. For example, generate videos that show how the human body works or how to perform a medical procedure.
```
```suggestion
```

I think this is a bit redundant as we mention how it can be used for educational purposes above.

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+T2V can be used to generate videos that can be used to help with disaster relief efforts. For example, a text to video model could be used to generate videos that show people how to stay safe during a hurricane or how to find food and water after a natural disaster.
+
+### Medical research
```
```suggestion
```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+
+### Voice overs and Speech
+
```
```suggestion
```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+### Voice overs and Speech
+
+
+T2V models can be used on digital humans or avatars for entertainment purposes. For example an AI newscaster to deliver daily news or for a filmmaker to create a short film or a music video.
```
```suggestion
Text-to-video models can be used to create an AI newscaster to deliver daily news, or for a film-maker to create a short film or a music video.

```

merveenoyan replied:
Maybe mention that for this it should be combined with a TTS model :')

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+T2V models can be used on digital humans or avatars for entertainment purposes. For example an AI newscaster to deliver daily news or for a filmmaker to create a short film or a music video.
+
+
+### Video summarization: 
```
```suggestion
### Video Summarization

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Video summarization: 
+
+T2V models can be used to summarize long videos into shorter, more concise videos. This can be useful for people who want to quickly get the gist of a video without having to watch the entire thing.
```
```suggestion
T2V models can be used to summarize long videos into shorter, more concise videos. This can be useful for people who want to quickly get the gist of a video without having to watch the entire thing.

```

Can you elaborate on how this can be done? I feel like this task would be video-to-video and not text-to-video.

snknitin replied:
My thought process here is that if we have a text-to-video model generate something long for say 10 minutes, based on a very detailed and descriptive prompt, being able to cut the prompt size to only include the basics can now generate a video that explains the same content but in 2 minutes with only the main points from the updated summary prompt

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Visual storytelling
+
+Text to video models can be used to create visual stories that can be used to communicate ideas and emotions. For example, the model could be used to create a video that tells the story of a refugee family or that shows the beauty of a natural landscape that might be difficult to capture under normal circumstances.
```
```suggestion
Text-to-video models can be used to create visual stories to communicate ideas and emotions. For example, the model could be used to create a video that tells the story of a refugee family, or a natural landscape that might be difficult to capture under normal circumstances.

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Music Visualization
+
+Creating music visualization is a complex, time-consuming, and resource-intensive process. Users define start and end prompts of music to parameterize the visualization and these prompts are warped between and generated according to the beat of the music for audioreactive video. Commonly used design patterns for improving generated videos are : "transitions", which express shifts in color, time,  ...
```
```suggestion
Creating music visualization is a complex, time-consuming, and resource-intensive process. Users define start and end prompts of music to parameterize the visualization and these prompts are warped between and generated according to the beat of the music for audio-reactive video. Commonly used design patterns for improving generated videos are "transitions", which express shifts in color, time, subject, or style, and "holds", which encourage visual emphasis and consistency. 
```
Last sentence can be divided, and can you add where the text-to-video models come into play?

snknitin replied:
Text, speech and music are all viewed as sequential encodings by the model. So text to video model is basically coming into play as an encoded representation of the music (between start to end prompt), and converted to video. I came across this idea for the use case from some youtube videos . We can remove it if not necessary

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Pose-Control
+
+Generating text-editable and pose-controllable character videos have a rising demand in creating digital humans/avatars. By utilizing easily obtained datasets (i.e.,image pose pair and pose-free video) and the pre-trained text-to-image (T2I) model to obtain the pose-controllable character videos, this learn a zero-initialized convolutional encoder to encode the pose information and then finetune  ...
```
```suggestion
Generating text-editable and pose-controllable character videos have a rising demand for creating avatars. By utilizing easily obtained datasets (i.e., image pose pair and pose-free video) and a text-to-image model to obtain the pose-controllable character videos, this learns a zero-initialized convolutional encoder to encode the pose information and then fine-tune the motion of the network via a pose-free video dataset.

```
Can you simplify this for a broader audience? 😊 

>  this learns a zero-initialized convolutional encoder to encode the pose information and then finetune the motion of the network via a pose-free video dataset.

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+## Task Variants
+
+There are different variants based on the inputs and outputs:
```
```suggestion
Text-to-video models have different variants based on inputs and outputs.

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+There are different variants based on the inputs and outputs:
+
+### Text to Video Retrieval
```
```suggestion
### Text-to-video Retrieval

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+This is a task that involves finding videos that are relevant to a given text query. This task is challenging because it requires the model to understand the meaning of the text query and to match it to the visual content of the videos. The best approach to text-to-video retrieval depends on the specific application. For example, if the application is to find videos that are relevant to a given t ...
+
+
+### Text-to-Video Editing
```
```suggestion
### Text-to-video Editing

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Text-to-Video Editing
+
+One T2V model variant is text-based video style and local attribute editing from the trained text-to-image model. Text to video editing models can make it easier to perform tasks like cropping, stabilization, color correction,resising and audio editing etc.,  and they can also help to ensure that the tasks are performed consistently
```
```suggestion
One text-to-video task is generating text-based video style and local attribute editing. Text-to-video editing models can make it easier to perform tasks like cropping, stabilization, color correction, resizing and audio editing consistently.

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+One T2V model variant is text-based video style and local attribute editing from the trained text-to-image model. Text to video editing models can make it easier to perform tasks like cropping, stabilization, color correction,resising and audio editing etc.,  and they can also help to ensure that the tasks are performed consistently
+
+
+### Text to Video Search
```
```suggestion
### Text-to-video Search

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Text to Video Search
+
+Text to video search is a task of retrieving videos that are relevant to a given text query. This can be challenging, as videos are a complex medium that can contain a lot of information. By using semantic analysis to extract the meaning of the text query, visual analysis to extract features from the videos, such as the objects and actions that are present in the video, and temporal analysis to c ...
```
```suggestion
Text-to-video search is the task of retrieving videos that are relevant to a given text query. This can be challenging, as videos are a complex medium that can contain a lot of information. By using semantic analysis to extract the meaning of the text query, visual analysis to extract features from the videos, such as the objects and actions that are present in the video, and temporal analysis to categorize relationships between the objects and actions in the video, we can determine which videos are most likely to be relevant to the text query.

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Text-driven Video Prediction
+
+Video Prediction is the task of predicting future frames given past video frames. Text-driven video prediction (TVP) is a task that involves generating a video sequence from a text description. The text description can be anything from a simple sentence to a detailed story. The goal of TVP is to generate a video that is both visually realistic and semantically consistent with the text description ...
```
```suggestion
Text-driven video prediction is the task of generating a video sequence from a text description. Text description can be anything from a simple sentence to a detailed story. The goal of this task is to generate a video that is both visually realistic and semantically consistent with the text description.

```

How is this different from text-to-video generation? I think this is the main task here.

snknitin replied:
I wrote it in a generic way but I wanted to frame this as a fine tuning of text to video models. Similar to how a stable diffusion or LLM can be trained on a global data and then finetuned to a niche sub data, text to video model can consume specific video series for finetuning and then predict what the next output of a text prompt would look like if it included connections from the fine tuning dataset. One application would be generating new episodes of a TV show

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+Video Prediction is the task of predicting future frames given past video frames. Text-driven video prediction (TVP) is a task that involves generating a video sequence from a text description. The text description can be anything from a simple sentence to a detailed story. The goal of TVP is to generate a video that is both visually realistic and semantically consistent with the text description ...
+
+
+### Video translation: 
```
```suggestion
### Video Translation

```

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+
+### Video translation: 
+
+Text-to-video models can be used to translate videos from one language to another or allow to query the multilingual text-video model with non-English sentences. This can be useful for people who want to watch videos in a language that they don't understand, especailly when multi-lingual captions are available for training.
```
```suggestion
Text-to-video translation models can translate videos from one language to another or allow to query the multilingual text-video model with non-English sentences. This can be useful for people who want to watch videos in a language that they don't understand, especially when multi-lingual captions are available for training.

```

Interesting, how is the input text and output video here? Can you explain?

snknitin replied:
There is a latent embedding space where the text input in English is converted to a representation that translates to concepts used to generate videos. If we are training a model to generate video from an English caption, we could expand the dataset to have multiple language captions per video to form a larger vocabulary that has artificial code switching(replacing some random repeating words into other language vocab) and this helps borrows statistical properties and distributions from higher resource language like English , to some lower resource language where data or text annotation isn't available, let's say Turkish. Obviously all concepts cannot be expressed in English and there are words in other languages like French or German to describe situations or objects. Searching or prompting for a video in any language should be able to pull up or generate a similar video about the same concept

snknitin replied:
There is a latent embedding space where the text input in English is converted to a representation that translates to concepts used to generate videos. If we are training a model to generate video from an English caption, we could expand the dataset to have multiple language captions per video to form a larger vocabulary that has artificial code switching(replacing some random repeating words into other language vocab) and this helps borrows statistical properties and distributions from higher resource language like English , to some lower resource language where data or text annotation isn't available, let's say Turkish. Obviously all concepts cannot be expressed in English and there are words in other languages like French or German to describe situations or objects. Searching or prompting for a video in any language should be able to pull up or generate a similar video about the same concept 

I was brainstorming use-cases . We can remove this if it doesn't align completely or sounds ambiguous.

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+In this area, you can insert useful resources about how to train or use a model for this task.
+
+
+## Most commonly used Datasets 
```
This section goes into data.ts, inside datasets :)

snknitin replied:
I'll move it !

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,115 @@
+* **DiDeMo** : The Distinct Describable Moments (DiDeMo) dataset is one of the largest and most diverse datasets for the temporal localization of events in videos given natural language descriptions. The videos are collected from Flickr and each video is trimmed to a maximum of 30 seconds.
+
+
+## Most commonly used Models 
```
Same with this one, it goes into data.ts :) you can find it in placeholder

snknitin replied:
Could you link me to a placeholder

merveenoyan reviewed (commented):
Very neat PR! I left some comments :) thank you 🤗

snknitin commented:
@merveenoyan I made all the commits and moved the data and models to the other file and addressed all the comments. Let me know how to proceed on those 2-3 use cases and how to make PR from the draft

snknitin commented:
@merveenoyan Can you let me know how to turn this draft PR into a full one? Please check the data.ts file as well

Muskan149 commented:
hi @snknitin, i can't find the text-to-video folder on hf's hub-docs. why is that so? sorry, i'm new to github contributions :)

snknitin commented:
> hi @snknitin, i can't find the text-to-video folder on hf's hub-docs. why is that so? sorry, i'm new to github contributions :)

Hi @Muskan149 , I think this PR is currently under review. Once approved it should be visible, if I'm not wrong. But i think you can fork the hubdocs repo and create a folder by the same name and work on your part with the same file structure and add stuff in the about.md file.  The changes can be merged once you merge the Pull request. @merveenoyan Please correct me if i'm wrong. This is my first PR too

tescao commented:
@Muskan149 @snknitin hi guys, same problem - I'm ready to contribute my part, but I think we'd better wait for the initial PR to be approved and merged so that we can pull that file and edit it. If each of us creates the file from scratch and PRs their own part, there's going to be a merge conflict as we'll be editing same lines in different ways, hence more work for everyone later on.

snknitin commented:
> @Muskan149 @snknitin hi guys, same problem - I'm ready to contribute my part, but I think we'd better wait for the initial PR to be approved and merged so that we can pull that file and edit it. If each of us creates the file from scratch and PRs their own part, there's going to be a merge conflict as we'll be editing same lines in different ways, hence more work for everyone later on.

Yes, I've made the first PR. Waiting for review from @merveenoyan

osanseviero commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,92 @@
+		},
+        {
+			description: "A multi-modal AI system that can generate novel videos with text, images, or video clips",
+			id:          "Runway Gen2",
```
The model IDs need to be valid hf.co/model repos

snknitin replied:
Modified it. This reduced the number of available models to just 2 though. Earlier I researched a couple of model that do text2video, but not all are available on the repo

snknitin commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,92 @@
+	datasets: [
+        {
+			description: "(Microsoft Research Video to Text) is a large-scale dataset for the open domain video captioning, which consists of 10,000 video clips from 20 categories, and each video clip is annotated with 20 English sentences.There are about 29,000 unique words in all captions. The standard splits uses 6,513 clips for training, 497 clips for validation, and 2,990 clips for testing.",
+			id:          "MSR-VTT",
```
I fixed it . Removed the ones that are not available and added both the dataset and models that exist on hf.  some of the id's are <username>/<dataset_name> . Is that fine?

merveenoyan replied:
Yep it is!

merveenoyan commented on tasks/src/text-to-video/about.md:
```diff
@@ -0,0 +1,76 @@
+
+Text-to-video models can be used to create visual stories to communicate ideas and emotions. For example, the model could be used to create a video that tells the story of a refugee family, or a natural landscape that might be difficult to capture under normal circumstances.
+
+
```
Can you remove some of the newlines? You can be inspired by the style in NLP task pages :)

snknitin replied:
Done ! Both these changes have been completed

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+const taskData: TaskDataCustom = {
+	datasets: [
+        {
+			description: "(Microsoft Research Video to Text) is a large-scale dataset for the open domain video captioning, which consists of 10,000 video clips from 20 categories, and each video clip is annotated with 20 English sentences.There are about 29,000 unique words in all captions. The standard splits uses 6,513 clips for training, 497 clips for validation, and 2,990 clips for testing.",
```
These are a bit too long, they should fit into single line, e.g. see Notes under datasets in this page: https://huggingface.co/tasks/image-segmentation

snknitin replied:
Done. Shortened them to a single sentence with brief description

merveenoyan reviewed (commented):
Can you also add a task summary? They're single lines appearing on top of every page. I gave very general feedbacks, happy to merge after they're addressed :)

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -3,31 +3,31 @@ import type { TaskDataCustom } from "../Types";
 	datasets: [
         {
-			description: "(Microsoft Research Video to Text) is a large-scale dataset for the open domain video captioning, which consists of 10,000 video clips from 20 categories, and each video clip is annotated with 20 English sentences.There are about 29,000 unique words in all captions. The standard splits uses 6,513 clips for training, 497 clips for validation, and 2,990 clips for testing.",
+			description: "(Microsoft Research Video to Text) is a large-scale dataset for the open domain video captioning",
```
```suggestion
			description: "Microsoft Research Video to Text is a large-scale dataset for open domain video captioning",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -3,31 +3,31 @@ import type { TaskDataCustom } from "../Types";
 		},
         {
-			description: "(UCF101 Human Actions dataset) is an extension of UCF50 and consists of 13,320 video clips, which are classified into 101 categories. These 101 categories can be classified into 5 types (Body motion, Human-human interactions, Human-object interactions, Playing musical instruments and Sports). The total length of these video clips is over 27 hours. All the videos are collected fro ...
+			description: "(UCF101 Human Actions dataset) consists of 13,320 video clips from youtube, which are classified into 101 categories.",
```
```suggestion
			description: "UCF101 Human Actions dataset consists of 13,320 video clips from YouTube, with 101 classes.",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -3,31 +3,31 @@ import type { TaskDataCustom } from "../Types";
 		},
         {
-			description: "The 20BN-SOMETHING-SOMETHING V2 dataset is a large collection of labeled video clips that show humans performing pre-defined basic actions with everyday objects.  It contains 220,847 videos, with 168,913 in the training set, 24,777 in the validation set and 27,157 in the test set. There are 174 labels.",
+			description: "The 20BN-SOMETHING-SOMETHING V2 dataset of labeled(174) video clips that show humans performing pre-defined basic actions with everyday objects.",
```
```suggestion
			description: "A dataset consisting of labeled video clips that show humans performing pre-defined basic actions with everyday objects.",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+			id:          "HuggingFaceM4/webvid",
+		},
+        {
+			description: "136M video clips with captions sourced from 1.2M Youtube videos (15 years of video) that describe complex tasks with step-by-step information",
```
```suggestion
			description: "Large dataset of video clips with captions sourced from Youtube videos on complex tasks with step-by-step information.",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+			id:          "totuta/youtube_subs_howto100M",
+		},
+        {
+			description: "The Distinct Describable Moments (DiDeMo) has 30 sec Flickr videos for the temporal localization of events in videos given descriptions.",
```
```suggestion
			description: "A dataset of short Flickr videos for the temporal localization of events with descriptions.",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+			id:          "PAIR/text2video-zero-controlnet-canny-arcane",
+		},
+		{
+			description: "Large-scale Pretraining for Text-to-Video Generation via Transformers",
```
```suggestion
			description: "A robust model for text-to-video generation.",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+	metrics:       [],
+	models:        [
+		{
+			description: "Text-to-Image Diffusion Models are Zero-Shot Video Generators",
```
```suggestion
			description: "A strong model for video generation.",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+        
+    ],
+	spaces:        [],
+	summary:       "",
```
Can you please add summary? It's the description that appears on top of the page :)

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+			id:          "quchenyuan/UCF101-ZIP",
+		},
+        {
+			description: "Kinetics Human Action Video Dataset is a large-scale, high-quality dataset for human action recognition in Youtube videos.",
```
```suggestion
			description: "A high-quality dataset for human action recognition in YouTube videos.",
```

merveenoyan reviewed (commented):
I've mostly done nitpicking to even shorten descriptions, thanks a lot!

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+        
+    ],
+	spaces:        [],
+	summary:       "Generates videos from input text. These models can be used to generate temporally and spatially consistent sequence of images based on text prompts.",
```
```suggestion
	summary:       "Text-to-video is the task of generating videos from text. ",
```

merveenoyan commented on tasks/src/text-to-video/data.ts:
```diff
@@ -0,0 +1,57 @@
+			id:          "nateraw/kinetics",
+		},
+        {
+			description: "A dataset consisting of labeled video clips that show humans performing pre-defined basic actions with everyday objects.",
```
```suggestion
			description: "A dataset of video clips of humans performing pre-defined basic actions with everyday objects.",
```

merveenoyan reviewed (approved):
I think we can merge! great work! 
I will add task schema and couple of other things and we can ship. What is your hugging face username? @snknitin

snknitin commented:
> I think we can merge! great work! I will add task schema and couple of other things and we can ship. What is your hugging face username? @snknitin

Awesome ! My hf username is [NikeZoldyck](https://huggingface.co/NikeZoldyck) . Thanks so much for this opportunity. This will be my first PR for 🤗
